explainer.md

```diff
@@ -215,6 +244,175 @@ third-party then signs either the exchanges in a package or the package as a
+`Content-Type` in the response, so the initial request is identical to any other
+request in the same context. It follows redirects, is constrained by the
+embedder's and any parent frame's Content Security Policy, and goes through the
+physical URL's Service Worker for navigations or the embedder's otherwise.
+
+Once the response comes back, its `Content-Type: application/signed-exchange`
+header tells the client that it represents a signed exchange, but it's initially
+treated like any other response: the Response object shown to the Service Worker
+(if any) is the bytes of the `application/signed-exchange` resource, not the
+response inside it, and the `application/signed-exchange` participates in caches
+the same way as any other resource.
+
+### Fetch the certificate chain
+
+The client parses the beginning of the `application/signed-exchange` resource to
+extract the [`Signature`
+header](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#signature-header).
+Each signature in the `Signature` header has a `certUrl` field that identifies a
+certificate chain to use to validate the signature, and each certificate chain
+is fetched and validated.
+
+The request for this certificate chain is made without credentials and skips Service
+Workers, for compatibility with browser network stack architectures. (This may
+change if network stacks prove to be less of a problem than expected.) The
+certificate is cached as a normal HTTP request.
+
+The client checks the leaf certificate's hash against the `certSha256` field of
+the `Signature` header and then validates the certificate chain. Certificate
+validation is a call to an OS-controlled library and may make further URL
+requests that aren't under the browser's control and therefore use a separate,
+probably ephemeral, cookie jar, cache, and set of service workers.
+
+Each signature with a valid certificate chain is passed on to the next step.
+
+### Signature verification
+
+Once the certificates are validated and enough of the
+`application/signed-exchange` resource is received to parse the claimed
+signed-exchange headers, the client extracts the logical URL from those headers
+and then tries to find a valid signature over the headers that is trusted for
+the claimed origin. If none of the signatures are valid, it either
+
+1. redirects to the logical URL as if the whole signed exchange were a 302
+   response, or
+2. fails with a network error.
+
+We're not yet certain which behavior is best. The first is slightly more
+resilient to clock skew, while the second encourages intermediates to get their
+implementations right.
+
+If the client does find a valid signature, it "releases" the headers for further
+processing and starts validating
+[mi-sha256](https://tools.ietf.org/html/draft-thomson-http-mice-02#section-2)
+records into a response stream as they arrive.
+
+### Prefetching stops here
```

Comment:
Looking at the comments by mnot, it seems intermediates may just go on and can cache the responses in the signed exchanges regardless of the request was for prefetches or not. Is that right, and if so should we note something here?

Reply:
I _think_ @mnot is talking about proxies caching the enveloped signed exchange instead of proxies unwrapping the signed exchange and caching its contents (https://github.com/WICG/webpackage/pull/173#discussion_r181182126), but if he does think they'll blindly unwrap things, I need to think about the implications.